metrics/metrics.go

```diff
@@ -0,0 +1,84 @@
+/*
+Package metrics provides a way to track metrics in the application. Functionality is strictly tailored to the needs of the package and is not meant to be a general purpose metrics library.
+*/
+package metrics
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Metrics is a struct that contains all the metrics that are used to track in the package
+type Metrics struct {
+	// collector is used to register the metrics
+	collector Collector
+
+	// Stats contains all the gauges that track the db-calculated stats
+	stats *prometheus.GaugeVec
+
+	// the histograms that track the time it takes to perform certain operations
+	verifyMerkleRoots *prometheus.HistogramVec
+	recordTransaction *prometheus.HistogramVec
+	queryTransaction  *prometheus.HistogramVec
+
+	// each cronJob is observed by the duration it takes to execute and the last time it was executed
+	cronHistogram     *prometheus.HistogramVec
+	cronLastExecution *prometheus.GaugeVec
+}
+
+// NewMetrics is a constructor for the Metrics struct
+func NewMetrics(collector Collector) *Metrics {
+	return &Metrics{
+		collector:         collector,
+		stats:             collector.RegisterGaugeVec(statsGaugeName, "name"),
+		verifyMerkleRoots: collector.RegisterHistogramVec(verifyMerkleRootsHistogramName, "classification"),
+		recordTransaction: collector.RegisterHistogramVec(recordTransactionHistogramName, "classification", "strategy"),
+		queryTransaction:  collector.RegisterHistogramVec(queryTransactionHistogramName, "classification"),
+		cronHistogram:     collector.RegisterHistogramVec(cronHistogramName, "name"),
+		cronLastExecution: collector.RegisterGaugeVec(cronLastExecutionGaugeName, "name"),
+	}
+}
+
+// EndWithClassification is a function returned by Track* methods that should be called when the tracked operation is finished
+type EndWithClassification func(success bool)
+
+// TrackVerifyMerkleRoots is used to track the time it takes to verify merkle roots
+func (m *Metrics) TrackVerifyMerkleRoots() EndWithClassification {
+	start := time.Now()
+	return func(success bool) {
+		m.verifyMerkleRoots.WithLabelValues(classify(success)).Observe(time.Since(start).Seconds())
+	}
+}
+
+// TrackRecordTransaction is used to track the time it takes to record a transaction
+func (m *Metrics) TrackRecordTransaction(strategyName string) EndWithClassification {
+	start := time.Now()
+	return func(success bool) {
+		m.recordTransaction.WithLabelValues(classify(success), strategyName).Observe(time.Since(start).Seconds())
```

Comment:
### minor
Just for the sake of simplicity when querying I would add an additional label direction
and for example under `incoming` we would have both external_incoming and internal_incoming. 
And still keep the current possibility to distinguish between external and internal incoming transactions. 

If you find this like a complex change I can live without it, therefore I'm approving the PR.

Reply:
For nowI would prefer not to add the change you mentioned because it would require me to re-do local tests. But in the future when we will be doing actual dashboards - we could add this feature. 

At the moment, we have the capability to retrieve both incoming transaction types by executing a Prometheus query similar to the following:
```
bux_record_transaction_histogram{name=~".*incoming_tx"}
```